### PR TITLE
[Parent][MBL-14026] Persist current student

### PR DESCRIPTION
**Pull request by Blackchai**
Modified a few screens, specifically around course and assignment details, to no longer require student details.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1023976&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1023977&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/screens/events/event_details_screen.dart [hunk 1]
```diff
@@ -31,16 +32,14 @@ class EventDetailsScreen extends StatefulWidget {
   final ScheduleItem event;
   final String eventId;
 
-  // Course ID and Student name/id are used for messaging. The message FAB will not be shown if any of these are null.
+  // Course ID and id are used for messaging. The message FAB will not be shown if any of these are null.
```
Comment (Blackchai) [inline apps/flutter_parent/lib/screens/events/event_details_screen.dart]: woops, another place I need to strip out student id.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1023978&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/screens/dashboard/dashboard_screen.dart [hunk 2]
```diff
@@ -85,6 +85,7 @@ class DashboardState extends State<DashboardScreen> {
       _students = widget.students;
       _selectedStudent = _students.first;
       _selectedStudentNotifier.value = _selectedStudent;
+      ApiPrefs.setCurrentStudent(_students.first);
```
Comment (StephenBrough) [inline apps/flutter_parent/lib/screens/dashboard/dashboard_screen.dart]: So this means that everything will have to go through the dashboard, including deep links, right? Just double checking our intentions.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024299&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**